client/js/ifml/fromjson.js

```diff
@@ -6,197 +6,314 @@
 
 var _ = require('lodash'),
     utils = require('almost-joint').utils,
-    ifml = require('./').ifml;
+    ifml = require('./').ifml,
+    almost = require('almost'),
+    Rule = almost.Rule,
+    createRule = almost.createRule,
+    core = require('almost-core'),
+    createTransformer = require('almost').createTransformer,
+    AException = almost.Exception;
 
-function mapViewContainer(container) {
-    return new ifml.elements.ViewContainer({
-        id: container.id,
-        name: container.attributes.name,
-        'default': container.attributes.default,
-        landmark: container.attributes.landmark,
-        xor: container.attributes.xor
-    });
-}
+var elementRules = [
+    createRule(
+        function (element) {
+            return ['ifml.ViewContainer', 'ifml.ViewComponent', 'ifml.Event', 'ifml.Action'].includes(element.type);
```

Comment:
Can be simplified using https://github.com/B3rn475/almostjs-extend#type-checking

```javascript
return model.isPositionedElement(element);
```